src/webgpu/util/conversion.ts

```diff
@@ -979,108 +983,104 @@ export interface ScalarBuilder {
   (value: number): Scalar;
 }
 
-/** Create an AbstractFloat from a numeric value, a JS `number`. */
-export function abstractFloat(value: number): Scalar {
-  const arr = new Float64Array([value]);
-  return new Scalar(TypeAbstractFloat, arr[0], arr);
+/** Create a Scalar of `type` by storing `value` as an element of `workingDataArray` and retrieving it. */
+function scalarFromValue(
+  type: ScalarType,
+  workingDataArray: TypedArrayBufferView,
+  value: number
+): Scalar {
+  workingDataU32[1] = 0;
```

Comment:
I assume this is to zero the array before assigning to (the aliased) `workingDataArray`? A comment here would be good.